### ISSUE TITLE
Add param_to_hook_all_reduce parameter in HF Trainer

### ISSUE THREAD

**Issue by awsankur** [huggingface/transformers#36941]
### Feature request

PyTorch DistributedDataParallel has `param_to_hook_all_reduce` option which is missing in trainer.py. This is needed to overlap gradient synchronization with backward pass

https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html

### Motivation

This will make Multi node DDP training more efficient

### Your contribution

Maybe

**Comment by geronimi73** [comment]
"Maybe" 😅

**Comment by awsankur** [comment]
It was funny but had to create this in a hurry.

What do you think of the `param_to_hook_all_reduce` parameter? I see all other parameters PyTorch DDP except this one.

**Comment by geronimi73** [comment]
never used it, in fact didn't know about it till now

**Comment by Rocketknight1** [comment]
cc @muellerzr @sunmarc

**Comment by SunMarc** [comment]
Happy to add this arg in HF trainer. This requires first a change in accelerate.
